app/pages/settings/email.jsx

```diff
@@ -0,0 +1,289 @@
+            >
+              {[this.state.meta.page_count].forEach(p => <option key={p} value={p}>{p}</option>)}
+            </select> :
+            '?'
+          }
+        </nav>
+      );
+    }
+  }
+
+  sortPreferences(preferences) {
+    const order = [
+      'participating_discussions',
+      'followed_discussions',
+      'started_discussions',
+      'mentions', 'group_mentions',
+      'messages'
+    ];
+    return preferences.sort((a, b) => {
+      return order.indexOf(a.category) > order.indexOf(b.category);
+    });
+  }
+
+  render() {
+    return (
+      <div className="content-container">
+        <p>
+          <AutoSave resource={this.props.user}>
+            <span className="form-label">
+              Email address
+            </span>
+            <br />
+            <input
+              type="text"
+              className="standard-input full"
+              name="email"
+              value={this.props.user.email}
+              onChange={handleInputChange.bind(this.props.user)}
+            />
+          </AutoSave>
+        </p>
+        <p><strong>Zooniverse email preferences</strong></p>
+        <p>
+          <AutoSave resource={this.props.user}>
+            <label>
+              <input
+                type="checkbox"
+                name="global_email_communication"
+                checked={this.props.user.global_email_communication}
+                onChange={handleInputChange.bind(this.props.user)}
+              />{' '}
+              Get general Zooniverse email updates
+            </label>
+          </AutoSave>
+          <br />
+          <AutoSave resource={this.props.user}>
+            <label>
+              <input
+                type="checkbox"
+                name="project_email_communication"
+                checked={this.props.user.project_email_communication}
+                onChange={handleInputChange.bind(this.props.user)}
+              />{' '}
+              Get email updates from the Projects you classify on
+            </label>
+          </AutoSave>
+          <br />
+          <AutoSave resource={this.props.user}>
+            <label>
+              <input
+                type="checkbox"
+                name="beta_email_communication"
+                checked={this.props.user.beta_email_communication}
+                onChange={handleInputChange.bind(this.props.user)}
+              />{' '}
+              Get beta project email updates and become a beta tester
+            </label>
+          </AutoSave>
+        </p>
+
+        <p><strong>Talk email preferences</strong></p>
+        <table className="talk-email-preferences">
+          <thead>
+            <tr>
+              <th>Send me an email</th>
+              <th>Immediately</th>
+              <th>Daily</th>
+              <th>Weekly</th>
+              <th>Never</th>
+            </tr>
+          </thead>
+          {this.renderTalkPreferences()}
```

Comment:
This will try to render a Promise as a React component, and throw the error you're seeing.